### PR TITLE
feat: add healthcheck docker

### LINKED ISSUE #682: docker: add healthcheck to docker image
### Describe the Feature Request

Would be great to create a healthcheck for Katana in the Dockerfile in order for other docker services to know when the RPC is ready 

### Describe Preferred Solution

Add similar strategy like this [pr](https://github.com/keep-starknet-strange/madara/pull/915/files) in Madara

### PR DESCRIPTION
**Pull request by ftupas**
Hi there, this smol pr adds healthcheck for docker

Fix #682

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -11,12 +11,20 @@ WORKDIR /dojo
...
+# Set an environment variable for the port
+ENV HEALTH_CHECK_PORT=5050
+
 FROM debian:buster-slim
 LABEL description="Dojo is a provable game engine and toolchain for building onchain games and autonomous worlds with Cairo" \
     authors="tarrence <tarrence@cartridge.gg>" \
     source="https://github.com/dojoengine/dojo" \
     documentation="https://book.dojoengine.org/"
 
+HEALTHCHECK --interval=10s --timeout=30s --start-period=10s --retries=10 \
```
Comment (Eikix) [inline Dockerfile:23]: If interval is 10s, and timeout is 30s, how can you retry 10 times? (im a docker noob)

Comment (ftupas) [reply Dockerfile:23]: Good catch! Actually we need to configure the start-period as well, we don't need 10s for the first healthcheck for katana

File: Dockerfile [hunk 2]
```diff
@@ -11,12 +11,20 @@ WORKDIR /dojo
...
+# Set an environment variable for the port
+ENV HEALTH_CHECK_PORT=5050
+
 FROM debian:buster-slim
 LABEL description="Dojo is a provable game engine and toolchain for building onchain games and autonomous worlds with Cairo" \
     authors="tarrence <tarrence@cartridge.gg>" \
     source="https://github.com/dojoengine/dojo" \
     documentation="https://book.dojoengine.org/"
 
+HEALTHCHECK --interval=3s --timeout=5s --start-period=1s --retries=5 \
```
Comment (ClementWalter) [inline Dockerfile:23]: isn't this only a `katana` healthcheck? I am not familiar with the dojo stack, but what if I use `sozo` as an entrypoint, whould this make sense?

Comment (ftupas) [reply Dockerfile:23]: That's a good point, I think if we want to have the healthcheck for `katana` only, we will need to make it in a separate Dockerfile. What do you think is the best approach here @tarrencev @kariy ?

**Review by tarrencev — APPROVED** [review]
lgtm. fwiw, katana exposes a health check on `GET /` which we use in k8s. fine with this approach though, since it mirrors madara

**Pull request merged**